src/set.ts

```diff
@@ -220,6 +220,15 @@ export class SuperchargedSet<T> implements Iterable<T> {
     return Array.from(this.set)
   }
 
+  reduce<U>(operation: (previous: U, current: T, set: SuperchargedSet<T>) => U, initial: U): U {
+    if (!initial && this.set.size === 0) throw new TypeError('Reduce of empty set with no initial value')

```

Comment:
I would like to split up this `if` statement. It mixes two states.

1. if the set is empty (you may use `this.isEmpty()`), just the initial value no matter what the initial value is
2. if the set is not empty and the initial value is not set: throw the error